### PR TITLE
chore(examples): add hello-python example using CompositeController

### PR DESCRIPTION
**Pull request by shovanmaity**
If you are using _'CompositeController'_ one basic misconfiguration can lead the controller to continuous reconciliations. When _'CompositeController'_ is used with the parent resource's status **undefined** as a sub-resource, then the reconciliation gets into an infinite loop. This happens since _'CompositeController'_ updates _'status.observedGeneration'_ with _'metadata.generation'_ as part of its reconciliation.

Following sums up the reconciliation logic when status is defined as a sub resource & vice-versa:

1. If parent's status is defined as a sub resource then reconciliation logic patches this status using status api end point. This does not change resource's `metadata.generation` field.
2. If parent's status is not defined as a sub resource then reconciliation logic updates the full parent object. This in turn leads to a update in parent's `metadata.generation` field. This leads to re-triggering of sync hook. In other words this forces the reconciliation to get into a never ending loop.

**Solution**: Add below to watch's CRD
```yaml
  subresources:
    status: {}
```

Signed-off-by: Shovan Maity <shovan.maity@mayadata.io>

### CODE REVIEW

File: examples/cctl/hello-python/readme.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  name: helloworlds.example.com
+spec:
+  group: example.com
+  version: v1
+  names:
+    kind: HelloWorld
+    plural: helloworlds
+    singular: helloworld
+  scope: Namespaced
+  subresources:
```
Comment (AmitKumarDas) [inline examples/cctl/hello-python/readme.md:48]: What's the difference between these two yamls

Comment (shovanmaity) [reply examples/cctl/hello-python/readme.md]: Forgot to remove. Updated thanks

File: examples/cctl/hello-python/readme.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
+## Example python Controller
+This example is taken from [metacontroller](https://metacontroller.app/guide/create/) doc. There is an issue in this example that leads to continuous reconciliations. When _'CompositeController'_ is used with the parent resource's status **undefined** as a sub-resource, then the reconciliation gets into an infinite loop. This happens since _'CompositeController'_ updates _'status.observedGeneration'_ with _'metadata.generation'_ as part of its reconciliation.
```
Comment (grzesuav) [inline examples/cctl/hello-python/readme.md:2]: I would rephrase this Paragraph. Now this gets blurry as description start with reference and go straight to exposing problem in original example and explaining how to fix it. I would see this as "standalone" example (without the need to go to other page) to understand what example want to show, with additional section emphasizing the necessity of `subresource: status`.

File: examples/cctl/hello-python/readme.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  group: example.com
+  version: v1
+  names:
+    kind: HelloWorld
+    plural: helloworlds
+    singular: helloworld
+  scope: Namespaced
+```
+
+### To try this example follow the below steps
```
Comment (grzesuav) [inline examples/cctl/hello-python/readme.md:56]: @AmitKumarDas maybe we should align to one way of presenting examples, like using `kind` ? I know that it can be re-used for minikube etc, but it would be good to either :
* do not refer to platform (skipping the kind/docker/minikube/setup) and explaining how to set up in different section of documentation. Therefore every example (using only `kubectl` command) will be uniersal
* agree that all examples should based on assumption of using `kind` //`minikube`

File: examples/cctl/hello-python/webhook.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    matchLabels:
+      app: hello-controller
+  template:
+    metadata:
+      labels:
+        app: hello-controller
+    spec:
+      containers:
+        - name: controller
+          image: python:2.7
```
Comment (grzesuav) [inline examples/cctl/hello-python/webhook.yaml:17]: maybe we can upgrade to python 3.8 as support for python 2 ended this year

File: examples/cctl/hello-python/readme.md (hunk 3 above)
Comment (grzesuav) [reply examples/cctl/hello-python/readme.md:56]: Actually created issue for that, https://github.com/AmitKumarDas/metac/issues/123 , so resolving the comment

File: examples/cctl/hello-python/readme.md (hunk 2 above)
Comment (AmitKumarDas) [reply examples/cctl/hello-python/readme.md:2]: @grzesuav I agree. Describing the example first and then an emphasis on status will make it easy for others to follow.

I am also wondering about docs. I feel it's time to have a separate site for metac as well.

**Comment by shovanmaity** [comment]
Hi @grzesuav @AmitKumarDas I have made some changes according to review comments. PTAL
